### ISSUE TITLE
Nested fields?

### ISSUE THREAD

**Issue by Maxpain177** [du5rte/graphql-mongodb-projection#5]
This package supports nested fields like `{ 'profile.name': true }`?

**Comment by du5rte** [comment]
@Maxpain177 not at the moment, but shouldn't be too hard to implement

**Comment by georgyfarniev** [comment]
Hello. Thanks for creating this nice library! Any updates about this part? For example, I use pagination, and I want to create projection not for root, but for nested query fragment. My query looks like this:
```
query {
  total
  data { field1, field }
}
```

I would like to create projection only for data subquery. How is it possible?

**Comment by natac13** [comment]
Hey @du5rte 

So I see this is an long standing issue and I was attempting to implement this feature since I would like to use it in a project I am currently working on. 

I believe I have the functionality done, however I would like to add tests to make sure. While I was trying to get my tests to pass I notice that the [MockDB](https://github.com/du5rte/graphql-mongodb-projection/blob/master/test/MongoDBMockup.js) was using the `user.json` file. So my projection as follows was not working.
```js
{
   _id: true,
   'nested.level': 1,
   'nested.nested.level': 1,
   'nested.nested.otherField': 1
}
```
FYI I added another user and updated the resolver to include my nested fields.

Therefore I am asking if it would be alright with you if I changed the MockDB to using the [MongoDB Jest setup](https://jestjs.io/docs/en/mongodb)? That way the resolver can query against mongodb directly, and the projection for the nested field will return the correct document shape.

**Comment by du5rte** [comment]
@natac13 Yeah, go for it! `MockDB` is a big nasty hack, definitely needs to be replaced

**Comment by natac13** [comment]
@du5rte Will do! Thanks
